src/metrics/transform.ts

```diff
@@ -0,0 +1,306 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  Histogram,
+  MetricData,
+  DataPoint,
+  DataPointType,
+  ExponentialHistogram,
+  ResourceMetrics,
+} from '@opentelemetry/sdk-metrics';
+import {MonitoredResource} from '@google-cloud/opentelemetry-resource-util';
+import * as path from 'path';
+import {MetricKind, ValueType} from './external-types';
+import {monitoring_v3} from 'googleapis';
+import {PreciseDate} from '@google-cloud/precise-date';
+import {
+  GAX_METER_NAME,
+  NATIVE_METRICS_PREFIX,
+  SPANNER_RESOURCE_TYPE,
+  METRIC_LABELS,
+  MONITORED_RESOURCE_LABELS,
+  METRIC_NAMES,
+} from './constants';
+
+/** Transforms a OpenTelemetry instrument type to a GCM MetricKind. */
+function transformMetricKind(metric: MetricData): MetricKind {
+  switch (metric.dataPointType) {
+    case DataPointType.SUM:
+      return metric.isMonotonic ? MetricKind.CUMULATIVE : MetricKind.GAUGE;
+    case DataPointType.GAUGE:
+      return MetricKind.GAUGE;
+    case DataPointType.HISTOGRAM:
+    case DataPointType.EXPONENTIAL_HISTOGRAM:
+      return MetricKind.CUMULATIVE;
+    default:
+      exhaust(metric);
+      // No logging needed as it will be done in transformPoints()
+      return MetricKind.UNSPECIFIED;
+  }
+}
+
+/** Transforms resource to Google Cloud Monitoring monitored resource */
+function transformResource(labels: {[key: string]: string}): MonitoredResource {
+  return {
+    type: SPANNER_RESOURCE_TYPE,
+    labels: labels,
+  } as MonitoredResource;
+}
+
+/** Transforms a OpenTelemetry ValueType to a GCM ValueType. */
+function transformValueType(metric: MetricData): ValueType {
+  const {
+    dataPointType,
+    descriptor: {name},
+  } = metric;
+
+  if (
+    dataPointType === DataPointType.HISTOGRAM ||
+    dataPointType === DataPointType.EXPONENTIAL_HISTOGRAM
+  ) {
+    return ValueType.DISTRIBUTION;
+  } else if (name.includes('count')) {
+    return ValueType.INT64;
+  }
+  console.info('Encountered unexpected metric %s', name);
+  return ValueType.VALUE_TYPE_UNSPECIFIED;
+}
+
+/**
+ * Convert the metrics data to a list of Google Cloud Monitoring time series.
+ */
+export function transformResourceMetricToTimeSeriesArray({
+  scopeMetrics,
+}: ResourceMetrics): monitoring_v3.Schema$TimeSeries[] {
+  if (!scopeMetrics) return [];
+
+  return (
+    scopeMetrics
+      // Only keep those whose scope.name matches 'gax-nodejs'.
+      .filter(({scope: {name}}) => name === GAX_METER_NAME)
```

Comment:
I don't think we need this `GAX_METER_NAME` . Unlike Java in Node we will be adding metrics in Spanner Client not in `gax-nodejs` client. In that case we should use our SPANNER_METER_NAME instead.

Reply:
Changed to spanner-nodejs.